doc/driver.mld

```diff
@@ -155,12 +155,16 @@ let link_output = ref [ "" ]
 
 let generate_output = ref [ "" ]
 
+(* Record the commands executed, their running time and optionally the path to
+   the produced file. *)
 let commands = ref [ ]
 
-let run cmd =
-  let cmd_str = Cmd.to_string cmd in
-  commands := cmd_str :: !commands;
-  OS.Cmd.(run_out ~err:OS.Cmd.err_run_out cmd |> to_lines) |> get_ok
+let run ?output_file cmd =
+  let t_start = Sys.time () in
+  let r = OS.Cmd.(run_out ~err:OS.Cmd.err_run_out cmd |> to_lines) |> get_ok in
+  let t_end = Sys.time () in
+  commands := (cmd, t_end -. t_start, output_file) :: !commands;
+  r
```

Comment:
Using `Unix.gettimeofday` gives very different results, that seem more reasonable...

```suggestion
let run ?output_file cmd =
  let t_start = Unix.gettimeofday () in
  let r = OS.Cmd.(run_out ~err:OS.Cmd.err_run_out cmd |> to_lines) |> get_ok in
  let t_end = Unix.gettimeofday () in
  commands := (cmd, t_end -. t_start, output_file) :: !commands;
  r
```